processing/src/main/java/org/apache/druid/query/aggregation/any/NumericAnyVectorAggregator.java

```diff
@@ -0,0 +1,128 @@
+  // Second rightmost bit for is found check (0 for not found and 1 for found)
+  @VisibleForTesting
+  static final byte BYTE_FLAG_FOUND_MASK = 0x02;
+  private static final byte BYTE_FLAG_NULL_MASK = 0x01;
+  protected static final int FOUND_VALUE_OFFSET = Byte.BYTES;
+
+  private final boolean replaceWithDefault = NullHandling.replaceWithDefault();
+  protected final VectorValueSelector vectorValueSelector;
+
+  public NumericAnyVectorAggregator(VectorValueSelector vectorValueSelector)
+  {
+    this.vectorValueSelector = vectorValueSelector;
+  }
+
+  /**
+   * Initialize the buffer value given the initial offset position within the byte buffer for initialization
+   */
+  abstract void initValue(ByteBuffer buf, int position);
+
+  /**
+   * Place the primitive value in the buffer given the initial offset position within the byte buffer
+   * at which the current aggregate value is stored.
+   */
+  abstract void putValue(ByteBuffer buf, int position, int row);
+
+  /**
+   * Place the primitive null value in the buffer, fiven the initial offset position within the byte buffer
+   * at which the current aggregate value is stored.
+   */
+  abstract void putNonNullValue(ByteBuffer buf, int position, Object value);
+
+  @Override
+  public void init(ByteBuffer buf, int position)
+  {
+    buf.put(position, replaceWithDefault ? NullHandling.IS_NOT_NULL_BYTE : NullHandling.IS_NULL_BYTE);
+    initValue(buf, position);
+  }
+
+  @Override
+  public void aggregate(ByteBuffer buf, int position, int startRow, int endRow)
+  {
+    if ((buf.get(position) & BYTE_FLAG_FOUND_MASK) != BYTE_FLAG_FOUND_MASK) {
+      boolean[] nulls = vectorValueSelector.getNullVector();
+      // check if there are any nulls
+      if (nulls != null && startRow <= nulls.length) {
+        for (int i = startRow; i < endRow; i++) {
+          // And there is actually a null
+          if (nulls[i]) {
+            putNull(buf, position);
+            return;
+          }
+        }
+      }
+      // There are no nulls, so put a value from the value selector
+      putValue(buf, position, startRow);
+      buf.put(position, (byte) (BYTE_FLAG_FOUND_MASK | NullHandling.IS_NOT_NULL_BYTE));
+    }
+  }
+
+  @Override
+  public void aggregate(
+      ByteBuffer buf,
+      int numRows,
+      int[] positions,
+      @Nullable int[] rows,
+      int positionOffset
+  )
+  {
+    for (int i = 0; i < numRows; i++) {
+      int position = positions[i] + positionOffset;
+      int row = rows == null ? i : rows[i];
+      aggregate(buf, position, row, row);
+    }
+  }
+
+  @Override
+  public void close()
+  {
+    // No resources to cleanup.
+  }
+
+  protected boolean isValueNull(ByteBuffer buf, int position)
+  {
+    return (buf.get(position) & BYTE_FLAG_NULL_MASK) == NullHandling.IS_NULL_BYTE;
+  }
+
+  private void putNull(ByteBuffer buf, int position)
+  {
+    if (!replaceWithDefault) {
+      buf.put(position, (byte) (BYTE_FLAG_FOUND_MASK | NullHandling.IS_NULL_BYTE));
```

Comment:
may be wrong here but shouldn't it be `buf.put(position, (byte) (BYTE_FLAG_FOUND_MASK | BYTE_FLAG_NULL_MASK | NullHandling.IS_NULL_BYTE))`
I see the same thing being done in buffer aggregator but not able to see how that is right. 
There are no tests as well.

Reply:
never mind. I see why it will work. NullHandling.IS_NULL_BYTE is `(byte) 1`